statsmodels/survey/summary_stats.py

```diff
@@ -0,0 +1,612 @@
+                                                self.data) - jdata)
+            mh = np.sqrt((nh - 1) / nh)
+            fh = np.sqrt(1 - self.design.fpc)
+            jdata = fh[:, None] * mh[:, None] * jdata
+        else:
+            nh = self.design.rep_weights.shape[1]
+            print(nh)
+            mh = np.sqrt((nh - 1) / nh)
+            jdata *= mh
+        vcov = np.dot(jdata.T, jdata)
+
+        return est, np.sqrt(np.diag(vcov))
+
+
+class SurveyMean(SurveyStat):
+    """
+    Calculates the mean for each column.
+
+    Parameters
+    -------
+    design : SurveyDesign object
+    data : ndarray
+        nxp array of the data to calculate the mean on
+    method: string
+        User inputs whether to get bootstrap or jackknife SE
+
+    Attributes
+    ----------
+    data : ndarray
+        The data which to calculate the mean on
+    design :
+        Points to the SurveyDesign object
+    est : ndarray
+        The point estimates of the statistic, calculated on the columns
+        of data.
+    vcov : ndarray
+        The variance-covariance of the estimates.
+    pseudo : ndarray
+        The jackknife pseudo-values.
+    """
+    def __init__(self, design, data, mse=False, replicates=None, bsn=None):
+        super().__init__(design, mse)
+        self.data = np.asarray(data)
+        if self.design.se_method == "jack":
+            self.est, self.std = self._jackknife()
+        else:
+            self.est, self.std = self._bootstrap(replicates, bsn)
+
+    def _stat(self, weights):
+        """
+        Returns calculation of mean.
+
+        Parameters
+        ----------
+        weights : np.array
+            The weights used to calculate the mean, will either be
+            original design weights or recalculated weights via jk,
+            boot, etc
+
+        Returns
+        -------
+        An array containing the statistic calculated on the columns
+        of the dataset.
+        """
+
+        return np.dot(weights, self.data) / np.sum(weights)
+
+
+class SurveyTotal(SurveyStat):
+    """
+    Calculates the total for each column.
+
+    Parameters
+    -------
+    design : SurveyDesign object
+    data : ndarray
+        nxp array of the data to calculate the total on
+    method: string
+        User inputs whether to get bootstrap or jackknife SE
+
+    Attributes
+    ----------
+    data : ndarray
+        The data which to calculate the mean on
+    design :
+        Points to the SurveyDesign object
+    est : ndarray
+        The point estimates of the statistic, calculated on the columns
+        of data.
+    vcov : ndarray
+        The variance-covariance of the estimates.
+    pseudo : ndarray
+        The jackknife pseudo-values.
+    """
+    def __init__(self, design, data, replicates=None, mse=False, bsn=None):
+        super().__init__(design, mse)
+        self.data = np.asarray(data)
+
+        if self.design.se_method == "jack":
+            self.est, self.std = self._jackknife()
```

Comment:
But this should be vcov, or maybe you should return both vcov and stderr (so every subclass doesn't have to repeat the sqrt calculation)